src/main/java/edu/hawaii/its/api/service/OotbGroupingPropertiesService.java

```diff
@@ -151,13 +171,28 @@ public void setCurrentUser(String currentUser) {
         this.currentUser = currentUser;
     }
 
+    public List<OotbGrouping> getGroupings() {
+        return groupings;
+    }
+
+    public void setGroupings(List<OotbGrouping> groupings) {
+        this.groupings = groupings;
+    }
+
     // Update active user profile with the values from ui request
 
     public OotbActiveProfileResult updateActiveUserProfile(OotbActiveProfile ootbActiveProfile) {
-        List<OotbGrouping> groupings = ootbActiveProfile.getGroupings();
+        if (modificationTracker.contains(ootbActiveProfile.getUid())) {
+            setCurrentUser(ootbActiveProfile.getUid());
+            return null;
+        }
```

Comment:
Just a question, is the purpose of this to prevent making extra calls below? Since you are using a set, the user's current groupings would not change in this scenario:
1. updateActiveUserProfile gets called with profile 1 and the groupings are set
2. updateActiveUserProfile gets called with profile 2 and the groupings are set
3. updateActiveUserProfile gets called with profile 1 but the groupings do not change

Reply:
Yes, this logic is to prevent extra API call to update active profile after user tried specific active profile at least once before. Once user call API to update specific active profile, those active profile information will be stored in the beans, and the functions in the OotbGroupingPropertiesService will query or update the bean with the value of current user in the <private String currentUser;> in OotbGroupingPropertiesService.

Reply:
Ok that makes more sense. Could you rename `modificationTracker` to something more descriptive like `ootbActiveProfileUuids`?

Reply:
Sounds better and clear!